### PR TITLE
feature/CVSB-14215 - Tech record history bug fix

### PR DESCRIPTION
**Pull request by irina005**
Ticket:

https://jira.dvsacloud.uk/browse/CVSB-14215

### CODE REVIEW

File: src/app/pipes/FilterMultipleRecordsPipe.ts [hunk 1]
```diff
@@ -6,7 +6,7 @@ export class FilterMultipleRecordsPipe implements PipeTransform {
   transform(multipleRecordsList: any): any {
     if (multipleRecordsList) {
       multipleRecordsList.forEach((techRecordList) => {
-        if (techRecordList) {
+        if (techRecordList && !Array.isArray(techRecordList.techRecord[0])) {
```
Comment (alleypa) [inline src/app/pipes/FilterMultipleRecordsPipe.ts:9]: please rename techRecordList to vehichleTechRecord and simplify the expression too

File: src/app/pipes/FilterMultipleRecordsPipe.ts [hunk 2]
```diff
@@ -6,7 +6,7 @@ export class FilterMultipleRecordsPipe implements PipeTransform {
   transform(multipleRecordsList: any): any {
```
Comment (alleypa) [inline src/app/pipes/FilterMultipleRecordsPipe.ts:6]: use of any

File: src/app/pipes/OrderByStatusPipe.ts [hunk 3]
```diff
@@ -6,6 +6,7 @@ export class OrderByStatusPipe implements PipeTransform {
 
   transform(techRecordList: any): any {
```
Comment (alleypa) [inline src/app/pipes/OrderByStatusPipe.ts:7]: use of any.

Comment (alleypa) [inline src/app/pipes/OrderByStatusPipe.ts:7]: remove any. Use correct types

File: src/app/pipes/FilterMultipleRecordsPipe.spec.ts [hunk 4]
```diff
@@ -54,21 +54,21 @@ describe('FilterMultipleRecordsPipe', () => {
     expect(vTechRecordList).toMatchSnapshot();
   });
 
-  it('if it CURRENT techRec it should set the technical record having status CURRENT for sorting', () => {
+  it('if it has CURRENT techRec it should set the technical record having status CURRENT for sorting', () => {
```
Comment (alleypa) [inline src/app/pipes/FilterMultipleRecordsPipe.spec.ts:57]: start your test description with 'should...'

File: src/app/pipes/FilterMultipleRecordsPipe.ts [hunk 5]
```diff
@@ -1,47 +1,46 @@
...
+    return vehicleTechnicalRecords;
+  }
+
+  getRecord(vehicleTechRecord: VehicleTechRecordModel): TechRecord {
+    let record: TechRecord;
+
+    record = vehicleTechRecord.techRecord.find(
+      (tRec: TechRecord) => tRec.statusCode === 'current'
+    );
+    if (record !== undefined) {
```
Comment (m-ujaffer) [inline src/app/pipes/FilterMultipleRecordsPipe.ts:29]: can be simplified to   ` if (record) `

File: src/app/pipes/FilterMultipleRecordsPipe.ts [hunk 6]
```diff
@@ -1,47 +1,46 @@
...
+      return record;
     }
+
+    record = vehicleTechRecord.techRecord.find(
+      (tRec: TechRecord) => tRec.statusCode === 'provisional'
+    );
+    if (record !== undefined) {
```
Comment (m-ujaffer) [inline src/app/pipes/FilterMultipleRecordsPipe.ts:36]: same as above

File: src/app/pipes/FilterMultipleRecordsPipe.ts [hunk 7]
```diff
@@ -1,47 +1,46 @@
...
+      return record;
+    }
+
```
Comment (m-ujaffer) [inline src/app/pipes/FilterMultipleRecordsPipe.ts:39]: can be further wrapped in to a function with `statusCode` as the parameter

**Pull request merged**